frontend/src/components/Icon.jsx

```diff
@@ -1,25 +1,32 @@
+// src/components/Icon.jsx
 import React from 'react';
 import { ICONS } from '../config/icons';
 
-const Icon = ({ name, size = 4, className = '', disableInvert = false, ...props }) => {
+const Icon = ({ name, size = 4, className = '', ...props }) => {
   const icon = ICONS[name];
-  
   if (!icon) {
-    console.warn(`Icon "${name}" not found in ICONS configuration`);
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(`Icon "${name}" not found in ICONS`);
+    }
     return null;
   }
 
   const sizeClass = `w-${size} h-${size}`;
-  const invertClass = disableInvert ? '' : 'dark:invert';
-  
+
+  // Render a masked span that inherits text color
   return (
-    <img
-      src={icon.src}
-      alt={icon.alt}
-      className={`${sizeClass} ${invertClass} ${className}`}
+    <span
+      role="img"
+      aria-label={icon.alt}
+      className={`inline-block align-middle ${sizeClass} ${className}`}
+      style={{
+        WebkitMask: `url(${icon.src}) no-repeat center / contain`,
+        mask: `url(${icon.src}) no-repeat center / contain`,
+        backgroundColor: 'currentColor',
+      }}
```

Comment:
Interesting. So this basically just makes the svg the same color as the text? Nice, it was a bit finicky to have to use the disableInvert flag sometimes.

Reply:
No I think this is better (if it works, I haven't tested myself yet), because the theme might have a light text-color even in lightmode in some cases (or vise versa).